### PR TITLE
Create a summary Markdown file for all asm diffs

### PR DESCRIPTION
**Pull request by BruceForstall**
Create a per-MCH file summary.md file, then accumulate them all into
a single overall summary Markdown file, for use in GitHub comments.
Uses the existing `jit-analyze --md` functionality.

Also, change asm diffs to not report missing data or asm diffs as a
replay failure.

Finally, don't overwrite superpmi.log files (or the new diff_summary.md files):
create new, unique file names for each run.

### CODE REVIEW

**Comment by BruceForstall** [comment]
@kunalspathak @dotnet/jit-contrib PTAL

File: src/coreclr/scripts/superpmi.py [hunk 1]
```diff
@@ -1804,8 +1824,12 @@ def replay_with_asm_diffs(self):
                         if return_code == 0:
                             logging.info("Clean SuperPMI replay")
                         else:
-                            files_with_replay_failures.append(mch_file)
                             result = False
+                            # Don't report as replay failure asm diffs (return code 2) or missing data (return code 3).
+                            # Anything else, such as compilation failure (return code 1, typically a JIT assert) will be
+                            # reported as a replay failure.
+                            if return_code != 2 and return_code != 3:
```
Comment (kunalspathak) [inline src/coreclr/scripts/superpmi.py:1831]: This is superpmi replay, so why would it get `return_code == 2` (asm diffs failure)?

File: src/coreclr/scripts/superpmi.py [hunk 2]
```diff
@@ -1971,8 +1998,32 @@ async def create_one_artifact(jit_path: str, location: str, flags) -> str:
...
+            if os.path.isfile(overall_md_summary_file):
+                os.remove(overall_md_summary_file)
+
+            with open(overall_md_summary_file, "w") as write_fh:
+                for summary_file_info in all_md_summary_files:
+                    summary_mch  = summary_file_info[0]
+                    summary_mch_filename = os.path.basename(summary_mch) # Display just the MCH filename, not the full path
+                    summary_file = summary_file_info[1]
+                    with open(summary_file, "r") as read_fh:
+                        write_fh.write(summary_mch_filename + ":\n\n")
```
Comment (kunalspathak) [inline src/coreclr/scripts/superpmi.py:2020]:
```suggestion
                        write_fh.write("##" + summary_mch_filename + ":" + os.linesep + os.linesep)
```

**Review by kunalspathak — APPROVED** [review]
Minor comments.

File: src/coreclr/scripts/superpmi.py (hunk 1 above)
Comment (BruceForstall) [reply src/coreclr/scripts/superpmi.py:1831]: This is the asm diffs case (only).

I'll go ahead and add similar "ignore" code to the `replay` case, but only for missing data.

File: src/coreclr/scripts/superpmi.py (hunk 2 above)
Comment (BruceForstall) [reply src/coreclr/scripts/superpmi.py:2020]: According to the docs (https://docs.python.org/3/library/os.html#os.linesep), you shouldn't use os.linesep when writing in text mode: Python auto-converts.

The `##` is a good idea.

**Pull request merged**